suprb/optimizer/solution/aco/builder.py

```diff
@@ -90,14 +90,14 @@ def relative_bounds_overlap(A: Rule, B: Rule) -> np.array:
     If one rule lies completely within the bounds of the other rule, the overlap is 1.
     """
 
-    intersections = (A.bounds[:, 0] <= B.bounds[:, 1]) & (B.bounds[:, 0] <= A.bounds[:, 1])
+    intersections = (A.input_space[:, 0] <= B.input_space[:, 1]) & (B.input_space[:, 0] <= A.input_space[:, 1])
```

Comment:
I dont think it is the input space that is referenced here. should be the actual matching function (I think)